dhis-2/dhis-web-api-test/src/test/java/org/hisp/dhis/webapi/controller/dataintegrity/AbstractDataIntegrityIntegrationTest.java

```diff
@@ -0,0 +1,241 @@
+        assertTrue( summary.isObject() );
+        assertEquals( expectedCount, summary.getCount() );
+        assertEquals( expectedPercentage, summary.getPercentage().intValue() );
+
+        postDetails( check );
+
+        JsonDataIntegrityDetails details = GET( "/dataIntegrity/" + check + "/details" )
+            .content()
+            .as( JsonDataIntegrityDetails.class );
+        assertTrue( details.exists() );
+        assertTrue( details.isObject() );
+        JsonList<JsonDataIntegrityDetails.JsonDataIntegrityIssue> issues = details.getIssues();
+        assertTrue( issues.exists() );
+        assertEquals( 1, issues.size() );
+        assertEquals( expectedUnit, issues.get( 0 ).getId() );
+        assertEquals( expectedName, issues.get( 0 ).getName() );
+
+        if ( expectedComment != null )
+        {
+            assertTrue( issues.get( 0 ).getComment().toString().contains( expectedComment ) );
+        }
+        assertEquals( issueType, details.getIssuesIdType() );
+    }
+
+    public final void DataIntegrityPositiveTestTemplate( String issueType, String check, Integer expectedCount,
+        Integer expectedPercentage, Set<String> expectedDetailsUnits,
+        Set<String> expectedDetailsNames, Set<String> expectedDetailsComments )
+    {
+
+        postSummary( check );
+        JsonDataIntegritySummary summary = GET( "/dataIntegrity/" + check + "/summary" )
+            .content()
+            .as( JsonDataIntegritySummary.class );
+        assertTrue( summary.exists() );
+        assertTrue( summary.isObject() );
+        assertEquals( expectedCount, summary.getCount() );
+        assertEquals( expectedPercentage, summary.getPercentage().intValue() );
+
+        postDetails( check );
+
+        JsonDataIntegrityDetails details = GET( "/dataIntegrity/" + check + "/details" )
+            .content()
+            .as( JsonDataIntegrityDetails.class );
+        assertTrue( details.exists() );
+        assertTrue( details.isObject() );
+        JsonList<JsonDataIntegrityDetails.JsonDataIntegrityIssue> issues = details.getIssues();
+        assertTrue( issues.exists() );
+
+        Set issueUIDs = issues.stream().map( issue -> issue.getId() ).collect( Collectors.toSet() );
+        assertEquals( issueUIDs, expectedDetailsUnits );
+
+        if ( expectedDetailsNames != null )
+        {
+            Set detailsNames = issues.stream().map( issue -> issue.getName() ).collect( Collectors.toSet() );
+            assertEquals( expectedDetailsNames, detailsNames );
+        }
+        if ( expectedDetailsComments != null )
+        {
+            Set detailsComments = issues.stream().map( issue -> issue.getComment() ).collect( Collectors.toSet() );
+            assertEquals( expectedDetailsNames, detailsComments );
+        }
+
+        assertEquals( issueType, details.getIssuesIdType() );
+    }
+
+    public final void DataIntegrityDivideByZeroTestTemplate( String detailsIssueType, String check )
```

Comment:
It is not clear from the name what this does. 
Maybe a name like `assertNoIssues`.